website/server/models/user/methods.js

```diff
@@ -148,18 +151,21 @@ schema.methods.sendMessage = async function sendMessage (userToReceiveMessage, o
   // Do not add the message twice when sending it to yourself
   let newSenderMessage;
 
+  const promises = [newReceiverMessage.save()];
+
   if (!sendingToYourself) {
     newSenderMessage = new Inbox({
       sent: true,
       ownerId: sender._id,
     });
     Object.assign(newSenderMessage, messageDefaults(senderMsg, userToReceiveMessage));
     setUserStyles(newSenderMessage, sender);
+    if (sender.flags.verifiedUsername === true) {
+      newSenderMessage.username = sender.auth.local.username;
```

Comment:
If I understand this code correctly....

When you send a PM to another player, this block of code puts a copy of your PM into your own inbox. That copy will contain the recipient's profile.name and User ID (not your profile.name).
Doesn't that mean that the username should be the recipient's username, not the sender's?

If that's true, I wonder if lines 163 to 165 (and lines 120 to 122) could be put into the messageDefaults() function. That would reduce a bit of code duplication and ensure that the same user's profile.name and username were always displayed together.

Reply:
I misunderstood how the user data is stored (Not sure why, since I've worked with that object in the past and wondered why it was modeled that way, so I should've known). I changed it and also moved the assignment to the `messageDefaults` function as suggested.